sw/host/cryptotest/testvectors/parsers/random_ecdsa_generator.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors.
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import argparse
+import json
+import logging
+import math
+import sys
+
+import jsonschema
+from Crypto.Hash import SHA256, SHA384, SHA512, SHA3_256, SHA3_384, SHA3_512
+from Crypto.PublicKey import ECC
+from Crypto.Signature import DSS
+from cryptotest_util import rng
+
+MESSAGE_LENGTH = 512
+
+
+def generate_test_vectors(args):
+    test_vectors = []
+    random = rng()
+    # Parse tests within the group
+    for count in range(args.count):
+        # Generate random ECC key pair
+        key_pair = ECC.generate(curve=args.curve, randfunc=random)
+        public = key_pair.pointQ
+        msg = random(MESSAGE_LENGTH)
+        signature = []
+        # Sign digest of message using private key
+        if args.hash_alg == "sha-256":
+            digest = SHA256.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha-384":
+            digest = SHA384.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha-512":
+            digest = SHA512.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha3-256":
+            digest = SHA3_256.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha3-384":
+            digest = SHA3_384.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
+        elif args.hash_alg == "sha3-512":
+            digest = SHA3_512.new(msg)
+            signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
+            signature = signer.sign(digest)
```

Comment:
nit: I think you can simplify this if/elif chain:
```suggestion
        if args.hash_alg == "sha-256":
            digest = SHA256.new(msg)
        elif args.hash_alg == "sha-384":
            digest = SHA384.new(msg)
        elif args.hash_alg == "sha-512":
            digest = SHA512.new(msg)
        elif args.hash_alg == "sha3-256":
            digest = SHA3_256.new(msg)
        elif args.hash_alg == "sha3-384":
            digest = SHA3_384.new(msg)
        elif args.hash_alg == "sha3-512":
            digest = SHA3_512.new(msg)
        signer = DSS.new(key_pair, 'fips-186-3', randfunc=random)
        signature = signer.sign(digest)
```

Reply:
I recall making this change locally at some point but it must have fallen through the cracks...